Scripts/Python/xPodium.py

```diff
@@ -0,0 +1,105 @@
+
+Additional permissions under GNU GPL version 3 section 7
+
+If you modify this Program, or any covered work, by linking or
+combining it with any of RAD Game Tools Bink SDK, Autodesk 3ds Max SDK,
+NVIDIA PhysX SDK, Microsoft DirectX SDK, OpenSSL library, Independent
+JPEG Group JPEG library, Microsoft Windows Media SDK, or Apple QuickTime SDK
+(or a modified version of those libraries),
+containing parts covered by the terms of the Bink SDK EULA, 3ds Max EULA,
+PhysX SDK EULA, DirectX SDK EULA, OpenSSL and SSLeay licenses, IJG
+JPEG Library README, Windows Media SDK EULA, or QuickTime SDK EULA, the
+licensors of this Program grant you additional
+permission to convey the resulting work. Corresponding Source for a
+non-source form of such a combination shall include the source code for
+the parts of OpenSSL and IJG JPEG Library used as well as that of the covered
+work.
+
+You can contact Cyan Worlds, Inc. by email legal@cyan.com
+ or by snail mail at:
+      Cyan Worlds, Inc.
+      14617 N Newport Hwy
+      Mead, WA   99021
+
+ *==LICENSE==* """
+
+from Plasma import *
+from PlasmaTypes import *
+
+actSwitch01 = ptAttribActivator(1, "act: Podium Button")
+respButtonOneshot = ptAttribResponder(2, "resp: Push Podium oneshot")
+respList = ptAttribResponderList(3, "ResponderList", byObject=1)
+stringFormat = ptAttribString(4, "Responder name format string")
+stringVarName = ptAttribString(5, "Age SDL variable name")
+
+class xPodium(ptResponder):
+
+    def __init__(self):
+        ptResponder.__init__(self)
+        self.id = 5245
+        version = 3
+        self.version = version
+        PtDebugPrint("__init__xPodium v.", version, ".0")
+
+    def OnFirstUpdate(self):
+        if not stringVarName.value:
+            PtDebugPrint("ERROR: xPodium.OnFirstUpdate():\tERROR: missing SDL var name")
+
+    def OnServerInitComplete(self):
+        ageSDL = PtGetAgeSDL()
+        if not PtGetPlayerList():
+            ageSDL[stringVarName.value] = (0,)
+        ageSDL.setNotify(self.key, stringVarName.value, 0.0)
+        ageSDL.sendToClients(stringVarName.value)
+        ageSDL.setFlags(stringVarName.value, 1, 1)
+
+    def OnNotify(self, state, id, events):
+        ageSDL = PtGetAgeSDL()
+        xPodiumSpeech = ageSDL[stringVarName.value][0]
+
+        if not state:
+            return
+
+        if id == actSwitch01.id:
+            respButtonOneshot.run(self.key, events=events)
+            return
+
+        elif id == respButtonOneshot.id and self.sceneobject.isLocallyOwned():
+            xPodiumSpeech -= 1
+            if xPodiumSpeech < 0:
+                xPodiumSpeech = len(respList.byObject) - 1
+            respName = (stringFormat.value % xPodiumSpeech)
+            if respName in respList.byObject:
+                ageSDL[stringVarName.value] = (xPodiumSpeech,)
+            else:
+                PtDebugPrint("ERROR: chisoPodium Invalid speech %d selected!" % xPodiumSpeech)
```

Comment:
```suggestion
                PtDebugPrint(f"ERROR: xPodium Invalid speech {xPodiumSpeech} selected!")
```
f-strings for great justice! Also missed the generic naming in this message.

Reply:
Oops! looks like I missed that one. Can ya tell I was playing around with the original a bit? 😝